cylc/flow/task_pool.py

```diff
@@ -1477,304 +1463,571 @@ def can_spawn(self, name: str, point: 'PointBase') -> bool:
-            is_manual_submit=is_manual_submit,
             flow_wait=flow_wait,
+            submit_num=submit_num,
+            transient=transient,
+            is_manual_submit=is_manual_submit
         )
-        if (name, point) in self.tasks_to_hold:
-            LOG.info(f"[{itask}] holding (as requested earlier)")
-            self.hold_active_task(itask)
-        elif self.hold_point and itask.point > self.hold_point:
-            # Hold if beyond the workflow hold point
-            LOG.info(
-                f"[{itask}] holding (beyond workflow "
-                f"hold point: {self.hold_point})"
-            )
-            self.hold_active_task(itask)
 
-        if self.stop_point and itask.point <= self.stop_point:
-            future_trigger_overrun = False
-            for pct in itask.state.prerequisites_get_target_points():
-                if pct > self.stop_point:
-                    future_trigger_overrun = True
-                    break
-            if future_trigger_overrun:
-                LOG.warning(
-                    f"[{itask}] won't run: depends on a task beyond "
-                    f"the stop point ({self.stop_point})"
-                )
-
-        # Satisfy any absolute triggers.
-        if (
-            itask.tdef.has_abs_triggers and
-            itask.state.prerequisites_are_not_all_satisfied()
-        ):
-            itask.state.satisfy_me(self.abs_outputs_done)
-
-        if flow_wait_done:
-            for outputs_str, fnums in (
-                self.workflow_db_mgr.pri_dao.select_task_outputs(
-                    itask.tdef.name, str(itask.point))
-            ).items():
+        if itask is not None:
+            # Update it with outputs that were already completed.
+            info = self.workflow_db_mgr.pri_dao.select_task_outputs(
+                itask.tdef.name, str(itask.point))
+            if not info:
+                self.db_add_new_flow_rows(itask)
+            spawn_kids = False
+            for outputs_str, fnums in info.items():
                 if flow_nums.intersection(fnums):
+                    if itask.flow_wait:
+                        spawn_kids = True
                     for msg in json.loads(outputs_str):
                         itask.state.outputs.set_completed_by_msg(msg)
-                    break
-            LOG.info(f"{itask} spawning on outputs after flow wait")
-            self.spawn_on_all_outputs(itask, completed_only=True)
-            return None
 
-        LOG.debug(f"[{itask}] spawned")
-        self.db_add_new_flow_rows(itask)
+            if spawn_kids:
+                self._spawn_after_flow_wait(itask)
+
         return itask
 
-    def force_spawn_children(
+    def _standardise_prereqs(self, prereqs: 'List[str]') -> 'List[Tokens]':
+        """Convert prerequisites to task output messages."""
+        _prereqs = []
+        for pre in [
+            Tokens(prereq, relative=True)
+            for prereq in (prereqs or [])
+        ]:
```

Comment:
```suggestion
        for prereq in prereqs:
            pre = Tokens(prereq, relative=True)
```
(If this breaks then the type annotation for `prereqs` is wrong)

Reply:
Ah yeah, the type changed several times during development.